cpu/stm32_common/periph/i2c_2.c

```diff
@@ -109,6 +109,14 @@ void i2c_init(i2c_t dev)
     gpio_init(i2c_config[dev].scl_pin, GPIO_OD_PU);
     gpio_init(i2c_config[dev].sda_pin, GPIO_OD_PU);
 #ifdef CPU_FAM_STM32F1
+    /* This is needed in case the remapped pins are used */
+    if (i2c_config[dev].scl_pin == GPIO_PIN(PORT_B, 8) ||
+        i2c_config[dev].sda_pin == GPIO_PIN(PORT_B, 9)) {
+        /* The remaping periph clock must first be enabled */
+        RCC->APB2ENR |= 0x00000001;
```

Comment:
(Very minor but) Is there a macro from the CMSIS that could be used instead of the hard coded bit position ?

Reply:
Yup, lazy me. All done!